pkg/logs/client/http/destination.go

```diff
@@ -347,12 +350,12 @@ func (d *Destination) updateRetryState(err error, isRetrying chan bool) bool {
 	}
 }
 
-func httpClientFactory(timeout time.Duration) func() *http.Client {
+func httpClientFactory(timeout time.Duration, cfg pkgconfigmodel.Reader) func() *http.Client {
 	return func() *http.Client {
 		return &http.Client{
 			Timeout: timeout,
 			// reusing core agent HTTP transport to benefit from proxy settings.
-			Transport: httputils.CreateHTTPTransport(pkgconfig.Datadog),
+			Transport: httputils.CreateHTTPTransport(cfg),
```

Comment:
It's unfortunate that we need to add some references to the global `pkgconfig.Datadog` in some places, but I'm glad to see that we're able to use the preferred `pkgconfigmodel.Reader` in spots like this.